.github/workflows/docker-build-test-upload.yml

```
@@ -1,78 +1,93 @@
-            image:
-                description: Image name
-                required: true
-                type: string
-            architecture:
-                description: Image architecture, e.g. amd64, arm64
-                required: true
-                type: string
-            runsOn:
-                description: GitHub Actions Runner image
-                required: true
-                type: string
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+    # only cancel in-progress jobs or runs for the current workflow - matches against branch & tags
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
 
 jobs:
     build-test-upload:
-        runs-on: ${{ inputs.runsOn }}
+        runs-on: ubuntu-latest
         continue-on-error: true
 
         steps:
             - name: Checkout Repo ⚡️
               uses: actions/checkout@v4
-            - name: Create dev environment 📦
-              uses: ./.github/actions/create-dev-env
+
+            - name: Login to Container Registry 🔑
+              uses: docker/login-action@v2
+              if: ${{ !github.event.pull_request.head.repo.fork }}
               with:
-                  architecture: ${{ inputs.architecture }}
+                  registry: ghcr.io
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
 
-            # Self-hosted runners share a state (whole VM) between runs
-            # Also, they might have running or stopped containers,
-            # which are not cleaned up by `docker system prun`
-            - name: Reset docker state and cleanup artifacts 🗑️
-              if: ${{ startsWith(inputs.runsOn, 'ubuntu') }}
-              run: |
-                  docker kill $(docker ps --quiet) || true
-                  docker rm $(docker ps --all --quiet) || true
-                  docker system prune --all --force
-                  rm -rf /tmp/aiidalab/
-              shell: bash
+            - name: Set up Docker Buildx
+              uses: docker/setup-buildx-action@v3
 
-            - name: Build image 🛠
-              working-directory: docker
-              run: docker buildx bake --set qe.platform=linux/${{ inputs.architecture }} -f docker-bake.hcl -f build.json --load
-              env:
-                  # Use buildx
-                  DOCKER_BUILDKIT: 1
-                  # Full logs for CI build
-                  BUILDKIT_PROGRESS: plain
-              shell: bash
+            - name: Docker meta 📝
+              if: ${{ !github.event.pull_request.head.repo.fork }}
+              id: meta
+              uses: docker/metadata-action@v4
+              with:
+                  images: |
+                      name=${{ env.IMAGE }}
+                  tags: |
+                      type=edge,enable={{is_default_branch}}
+                      type=ref,event=pr
+                      type=match,pattern=v(\d{2}\.\d{2}.\d+.*),group=1
```

```suggestion
                      type=match,pattern=v(\d{4}\.\d{4}.*),group=1
```

I guess you copy this from aiida-core? :smile:

---

Nope, I actually took it from this repo. :-P I think it is correct, no? The current QeApp version is `24.10.0a1`

---

Ah, true, my bad, I confused with aiidalab-docker-stack :sweat_smile: